### PR TITLE
Improve alerts

### PR DESCRIPTION
**Pull request by ZeHiro**
Add date of operation + Format depending on currency + Format date based on local

### CODE REVIEW

File: server/models/alert.js [hunk 1]
```diff
@@ -89,29 +91,33 @@ Alert.prototype.testBalance = function(balance) {
            (this.order === 'gt' && balance >= alertLimit);
 };
 
-Alert.prototype.formatOperationMessage = function(operation, accountName) {
+Alert.prototype.formatOperationMessage = function(operation, accountName, locale, accountCurrency) {
```
Comment (bnjbvr) [inline server/models/alert.js]: Can you make that the locale be the first argument, please? The order of arguments seems random like this.

File: server/lib/alert-manager.js [hunk 2]
```diff
@@ -45,16 +46,24 @@ ${$t('server.email.signature')}
...
+            let defaultCurrency = await Config.byName('defaultCurrency').value;
             // Map account to names
             let accounts = await Account.all();
-            let accountNames = new Map;
+            let accountsMap = new Map;
             for (let a of accounts) {
-                accountNames.set(a.accountNumber, a.title);
+                accountsMap.set(a.accountNumber, {
+                    title: a.title,
+                    currency: a.currency || defaultCurrency
```
Comment (bnjbvr) [inline server/lib/alert-manager.js]: how about directly storing the currency format function here instead? So that you can pass it to formatOperationMessage and there's no need to call makeFormat for each alert.

File: server/lib/alert-manager.js [hunk 3]
```diff
@@ -104,9 +117,9 @@ ${$t('server.email.signature')}
                 for (let alert of alerts) {
                     if (!alert.testBalance(balance))
                         continue;
-
+                    let currency = account.currency || defaultCurrency;
                     let text =
-                        alert.formatAccountMessage(account.title, balance);
+                        alert.formatAccountMessage(account.title, balance, currency);
```
Comment (bnjbvr) [inline server/lib/alert-manager.js]: Again, I'd prefer the format function to be passed to formatAccountMessage from the above, so from here. Maybe the format function can be created before this loop, too?

File: server/lib/report-manager.js [hunk 4]
```diff
@@ -110,7 +112,7 @@ class ReportManager
         content += '\n';
 
         for (let account of accounts) {
-            let lastCheck = moment(account.lastCheck).format('DD/MM/YYYY');
+            let lastCheck = moment(account.lastCheck).locale(locale).format('L');
```
Comment (bnjbvr) [inline server/lib/report-manager.js]: Any chance to hoist this line into an helper function, in the shared helpers? Specifically, we shouldn't need to load the locale each time...

**Comment by bnjbvr** [comment]
Thank you for this pull request! A few comments above.

File: server/models/alert.js (hunk 1 above)
Comment (ZeHiro) [inline server/models/alert.js]: locale is deleted

File: server/lib/alert-manager.js (hunk 2 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: yes

File: server/lib/alert-manager.js (hunk 3 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: yes

File: server/lib/report-manager.js (hunk 4 above)
Comment (ZeHiro) [inline server/lib/report-manager.js]: yes

File: server/helpers.js [hunk 5]
```diff
@@ -119,3 +121,13 @@ export function isCredentialError(err) {
            err.errCode === getErrorCode('INVALID_PARAMETERS') ||
            err.errCode === getErrorCode('NO_PASSWORD');
 }
+
+export function setupMoment(locale) {
+    if (locale) {
```
Comment (bnjbvr) [inline server/helpers.js]: Can you do this at the end of `setupTranslator`, instead?

File: server/lib/alert-manager.js [hunk 6]
```diff
@@ -72,14 +79,17 @@ ${$t('server.email.signature')}
                     continue;
                 }
 
-                let accountName = accountNames.get(operation.bankAccount);
+                // Set the account information
+                let account = accountsMap.get(operation.bankAccount);
+                let accountName = account.title;
+                let currencyFormatter = account.currencyFormatter;
```
Comment (bnjbvr) [inline server/lib/alert-manager.js]: `let { title: accountName, currencyFormatter } = accountsMap.get(operation.bankAccount);`

File: server/lib/alert-manager.js [hunk 7]
```diff
@@ -105,9 +118,11 @@ ${$t('server.email.signature')}
                     if (!alert.testBalance(balance))
                         continue;
 
+                    // Set the currency formatter
+                    let curr = account.currency || defaultCurrency;
+                    let currencyFormatter = currency.makeFormat(curr);
                     let text =
-                        alert.formatAccountMessage(account.title, balance);
-
+                        alert.formatAccountMessage(account.title, balance, currencyFormatter);
```
Comment (bnjbvr) [inline server/lib/alert-manager.js]: Now that the limit is to 100 chars, can this fit on the above line, next to `text =`?

File: server/lib/report-manager.js [hunk 8]
```diff
@@ -101,7 +99,7 @@ class ReportManager
         subject = $t('server.email.report.subject', { frequency });
         subject = `Kresus - ${subject}`;
 
-        let today = moment().format('DD/MM/YYYY');
+        let today = moment().format('L');
```
Comment (bnjbvr) [inline server/lib/report-manager.js]: Does this work? As `moment` is already an instance of the lib `moment`, I don't know whether we can again call it to create new instances based on the parent...

File: server/helpers.js (hunk 5 above)
Comment (ZeHiro) [inline server/helpers.js]: Hmm, setupTranslator is shared between client and server. But this should be done when we use moment (I plan to do so)

File: server/lib/alert-manager.js (hunk 7 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: Nope

File: server/lib/alert-manager.js (hunk 6 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: OK

File: server/lib/report-manager.js (hunk 8 above)
Comment (ZeHiro) [inline server/lib/report-manager.js]: The moment lib is not loaded in this module. It is loaded throught the helpers. Maybe this should also be done for all other module using moment. And yes, it does work

File: server/helpers.js (hunk 5 above)
Comment (bnjbvr) [inline server/helpers.js]: Out of curiosity: it does seem that the `locale` function doesn't return a new value, instead it sets something on the original variable `mom`. So I was wondering whether we really need to `export let moment = mom;`? Does it work if you don't do it?

File: server/lib/alert-manager.js [hunk 9]
```diff
@@ -45,11 +46,17 @@ ${$t('server.email.signature')}
 
     async checkAlertsForOperations(operations) {
         try {
+            // Default currency
```
Comment (bnjbvr) [inline server/lib/alert-manager.js]: nit: i'd remove the comment here.

File: server/lib/alert-manager.js [hunk 10]
```diff
@@ -93,6 +101,9 @@ ${$t('server.email.signature')}
 
     async checkAlertsForAccounts() {
         try {
+            // Default currency
```
Comment (bnjbvr) [inline server/lib/alert-manager.js]: ditto

File: server/helpers.js (hunk 5 above)
Comment (bnjbvr) [inline server/helpers.js]: Other idea: rather than leaking the `moment` value, could we have an exported function here, `formatDateToLocaleString(date) { return moment(date).format('L'); }`? This would hide the implementation detail that is `moment`, and in particular the `'L'` constant that is in several files.

File: server/lib/report-manager.js (hunk 8 above)
Comment (bnjbvr) [inline server/lib/report-manager.js]: Thanks, I misread the initial code. This makes sense.

File: server/helpers.js (hunk 5 above)
Comment (ZeHiro) [inline server/helpers.js]: could do that (final proposal. But I wanted to keep the possibility to manipulate the moments in an understandable way

File: server/lib/alert-manager.js (hunk 9 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: Yep

File: server/lib/alert-manager.js (hunk 10 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: Yep

**Comment by bnjbvr** [comment]
Hi! Is this PR ready for review? I see plenty of comments not addressed yet, but that might Github's UI not working perfectly here.

File: server/lib/alert-manager.js (hunk 9 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: Done

File: server/lib/alert-manager.js (hunk 10 above)
Comment (ZeHiro) [inline server/lib/alert-manager.js]: Done

File: server/lib/report-manager.js [hunk 11]
```diff
@@ -1,13 +1,14 @@
...
+import { makeLogger, KError, translate as $t, currency,
+formatDateToLocaleString } from '../helpers';
 import Emailer from './emailer';
 
 import Account   from '../models/account';
 import Alert     from '../models/alert';
 import Operation from '../models/operation';
 import Config    from '../models/config';
 
+import moment from 'moment';
```
Comment (bnjbvr) [inline server/lib/report-manager.js]: Why is this import needed here? It seems unused here.

Comment (ZeHiro) [inline server/lib/report-manager.js]: it is used 3 times. But before it was at the top of the file.

**Comment by bnjbvr** [comment]
Thanks!
